iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java

```diff
@@ -32,6 +32,11 @@ void isTrue() {
     assertThat(TreePredicates.isTrue().test(text("true"))).isTrue();
   }
 
+  @Test
+  void isEqualTo() {
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("VALUE_TEST"))).isTrue();
```

Comment:
Maybe you add another negative test, as well as a test where none TextTree is tested in the predicate. (Yes, I know the other predicate is missing this test as well. If possible, you might add it there as well)